0.38.0 introduces TFBuild obsolete warning for everyone not just users of property

In 0.38.0, everyone will get this wanting in 0.38.0, regardless if they're using it or not as it's a cached property it's invoked in the code gen.

```
 (2783,24): warning CS0618: 'BuildSystemAliases.TFBuild(ICakeContext)' is obsolete: 'Use AzurePipelines instead.'
````

--- a/build/version.cake
+++ b/build/version.cake
@@ -20,11 +20,11 @@ public class BuildVersion
         if (!parameters.SkipGitVersion)
         {
             // Temp Workaround GitVersion Azure Pipelines
-            var tfBuild = context.TFBuild();
+            var azurePipelines = context.AzurePipelines();
             string sourceBranch = string.Empty;
-            if ((tfBuild.IsRunningOnAzurePipelinesHosted || tfBuild.IsRunningOnAzurePipelines) && tfBuild.Environment.PullRequest.Number > 0)
+            if ((azurePipelines.IsRunningOnAzurePipelinesHosted || azurePipelines.IsRunningOnAzurePipelines) && azurePipelines.Environment.PullRequest.Number > 0)
             {
-                 sourceBranch = $"PullRequest{tfBuild.Environment.PullRequest.Number}";
+                 sourceBranch = $"PullRequest{azurePipelines.Environment.PullRequest.Number}";
                  context.Information("Overriding Azure Pipelines branch name with: {0}", sourceBranch);
             }
 

--- a/src/Cake.Core.Tests/Unit/Scripting/CodeGen/Expected/Properties/Cached_Obsolete_ExplicitWarning_WithMessage
+++ b/src/Cake.Core.Tests/Unit/Scripting/CodeGen/Expected/Properties/Cached_Obsolete_ExplicitWarning_WithMessage
@@ -1,4 +1,5 @@
 ﻿private System.Int32? _Cached_Obsolete_ExplicitWarning_WithMessage;
+[Obsolete("Please use Foo.Bar instead.")]
 public System.Int32 Cached_Obsolete_ExplicitWarning_WithMessage
 {
     get
@@ -6,7 +7,9 @@ public System.Int32 Cached_Obsolete_ExplicitWarning_WithMessage
         Context.Log.Warning("Warning: The alias Cached_Obsolete_ExplicitWarning_WithMessage has been made obsolete. Please use Foo.Bar instead.");
         if (_Cached_Obsolete_ExplicitWarning_WithMessage==null)
         {
+#pragma warning disable CS0618
             _Cached_Obsolete_ExplicitWarning_WithMessage = Cake.Core.Tests.Data.PropertyAliasGeneratorData.Cached_Obsolete_ExplicitWarning_WithMessage(Context);
+#pragma warning restore CS0618
         }
         return _Cached_Obsolete_ExplicitWarning_WithMessage.Value;
     }

--- a/src/Cake.Core.Tests/Unit/Scripting/CodeGen/Expected/Properties/Cached_Obsolete_ImplicitWarning_NoMessage
+++ b/src/Cake.Core.Tests/Unit/Scripting/CodeGen/Expected/Properties/Cached_Obsolete_ImplicitWarning_NoMessage
@@ -1,4 +1,5 @@
 ﻿private System.Int32? _Cached_Obsolete_ImplicitWarning_NoMessage;
+[Obsolete]
 public System.Int32 Cached_Obsolete_ImplicitWarning_NoMessage
 {
     get
@@ -6,7 +7,9 @@ public System.Int32 Cached_Obsolete_ImplicitWarning_NoMessage
         Context.Log.Warning("Warning: The alias Cached_Obsolete_ImplicitWarning_NoMessage has been made obsolete.");
         if (_Cached_Obsolete_ImplicitWarning_NoMessage==null)
         {
+#pragma warning disable CS0618
             _Cached_Obsolete_ImplicitWarning_NoMessage = Cake.Core.Tests.Data.PropertyAliasGeneratorData.Cached_Obsolete_ImplicitWarning_NoMessage(Context);
+#pragma warning restore CS0618
         }
         return _Cached_Obsolete_ImplicitWarning_NoMessage.Value;
     }

--- a/src/Cake.Core.Tests/Unit/Scripting/CodeGen/Expected/Properties/Cached_Obsolete_ImplicitWarning_WithMessage
+++ b/src/Cake.Core.Tests/Unit/Scripting/CodeGen/Expected/Properties/Cached_Obsolete_ImplicitWarning_WithMessage
@@ -1,4 +1,5 @@
 ﻿private System.Int32? _Cached_Obsolete_ImplicitWarning_WithMessage;
+[Obsolete("Please use Foo.Bar instead.")]
 public System.Int32 Cached_Obsolete_ImplicitWarning_WithMessage
 {
     get
@@ -6,7 +7,9 @@ public System.Int32 Cached_Obsolete_ImplicitWarning_WithMessage
         Context.Log.Warning("Warning: The alias Cached_Obsolete_ImplicitWarning_WithMessage has been made obsolete. Please use Foo.Bar instead.");
         if (_Cached_Obsolete_ImplicitWarning_WithMessage==null)
         {
+#pragma warning disable CS0618
             _Cached_Obsolete_ImplicitWarning_WithMessage = Cake.Core.Tests.Data.PropertyAliasGeneratorData.Cached_Obsolete_ImplicitWarning_WithMessage(Context);
+#pragma warning restore CS0618
         }
         return _Cached_Obsolete_ImplicitWarning_WithMessage.Value;
     }

--- a/src/Cake.Core/Scripting/CodeGen/PropertyAliasGenerator.cs
+++ b/src/Cake.Core/Scripting/CodeGen/PropertyAliasGenerator.cs
@@ -184,12 +184,9 @@ namespace Cake.Core.Scripting.CodeGen
 
             // Property is obsolete?
             var obsolete = method.GetCustomAttribute<ObsoleteAttribute>();
-            if (obsolete != null)
+            if (obsolete != null && obsolete.IsError)
             {
-                if (obsolete.IsError)
-                {
-                    return GenerateCode(method, out hash);
-                }
+                return GenerateCode(method, out hash);
             }
 
             // Backing field.
@@ -205,6 +202,19 @@ namespace Cake.Core.Scripting.CodeGen
             builder.AppendLine();
 
             // Property
+            if (obsolete != null)
+            {
+                if (string.IsNullOrEmpty(obsolete.Message))
+                {
+                    builder.Append("[Obsolete]");
+                }
+                else
+                {
+                    builder.AppendFormat("[Obsolete(\"{0}\")]", obsolete.Message);
+                }
+
+                builder.AppendLine();
+            }
             hash = GenerateCommonInitalCode(method, ref builder);
             builder.Append("{");
             builder.AppendLine();
@@ -227,10 +237,22 @@ namespace Cake.Core.Scripting.CodeGen
 
             builder.Append("        {");
             builder.AppendLine();
+
+            if (obsolete != null)
+            {
+                builder.AppendLine("#pragma warning disable CS0618");
+            }
+
             builder.AppendFormat("            _{0} = ", method.Name);
             builder.Append(method.GetFullName());
             builder.Append("(Context);");
             builder.AppendLine();
+
+            if (obsolete != null)
+            {
+                builder.AppendLine("#pragma warning restore CS0618");
+            }
+
             builder.Append("        }");
             builder.AppendLine();
             builder.AppendFormat("        return _{0}", method.Name);

